File: modules/pipelinetest/resources/pipeline/pipelines/timeout-exec-test.js
/*
 * Copyright (c) 2017-2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
 */
print('hello Nashorn timeout world!');
print("arguments.length: " + arguments.length);
var timeout = 10;
for (var i = 0; i < arguments.length; i++) {
    print("  arg[" + i  + "]=" + arguments[i]);
    if (arguments[i] === '-t') {
        timeout = parseInt(arguments[i+1]);
    }
}
print('sleeping for ' + timeout + ' seconds');
java.lang.Thread.sleep(1000 * timeout);
print('goodbye Nashorn timeout world!');

File: modules/pipelinetest/resources/pipeline/pipelines/TimeoutExecTest.java
/*
 * Copyright (c) 2020 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
 */
class TimeoutExecTest {
    public static void main(String[] args) throws Exception
    {
        println("hello java timeout world!");
        println("arguments.length: " + args.length);
        var timeout = 10;
        for (var i = 0; i < args.length; i++)
        {
            println("  arg[" + i + "]=" + args[i]);
            if ("-t".equals(args[i]))
            {
                timeout = Integer.parseInt(args[i+1]);
            }
        }

        println("sleeping for " + timeout + " seconds");
        Thread.sleep(1000 * timeout);
        println("goodbye java timeout world!");
    }

    public static void println(String s)
    {
        System.out.println(s);
    }
}
